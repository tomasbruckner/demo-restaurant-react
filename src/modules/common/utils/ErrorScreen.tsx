import React, { PureComponent } from 'react';
import PageLayout from '../layouts/PageLayout';
import { I18N_CORE } from 'core/i18n/i18n';
import { WithTranslation, withTranslation } from 'react-i18next';

export class ErrorScreen extends PureComponent<{ title: string } & WithTranslation> {
  render() {
    const { t } = this.props;

    return <PageLayout title={t('page.error.title')} />;
  }
}

export default withTranslation(I18N_CORE)(ErrorScreen);
