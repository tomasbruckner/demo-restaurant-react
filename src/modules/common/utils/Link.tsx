import { Link as RouterLink } from 'react-router-dom';

export default RouterLink;
