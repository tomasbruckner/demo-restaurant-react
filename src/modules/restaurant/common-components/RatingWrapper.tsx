import styled from 'styled-components/macro';
import React, { PureComponent } from 'react';
import Rating from 'modules/common/components/controls/Rating';
import { Typography, withStyles } from '@material-ui/core';
import { I18N_CORE } from 'core/i18n/i18n';
import { WithTranslation, withTranslation } from 'react-i18next';
import EditIcon from '@material-ui/icons/Edit';
import DeleteIcon from '@material-ui/icons/Delete';
import Button from 'modules/common/components/controls/Button';

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  max-width: 500px;
`;

const StyledButton = withStyles({
  root: {
    padding: '16px',
  },
})(Button);

type Props = {
  averageRating: number;
  numberOfReviews?: number;
  text?: string;
  canEdit?: boolean;
  onEdit?: () => void;
  onDelete?: () => void;
};

export class RatingWrapper extends PureComponent<Props & WithTranslation> {
  onEdit = () => {
    const { onEdit } = this.props;

    if (onEdit) {
      onEdit();
    }
  };

  onDelete = () => {
    const { onDelete } = this.props;

    if (onDelete) {
      onDelete();
    }
  };

  render() {
    const {
      averageRating,
      numberOfReviews,
      t,
      canEdit = true,
      text = t('component.restaurant-item.reviews', { numberOfReviews }),
    } = this.props;

    return (
      <Wrapper>
        <Rating value={averageRating} precision={0.5} readOnly size="large" />
        <Typography variant="subtitle1" color="textSecondary">
          {text}
        </Typography>
        {canEdit ? (
          <span>
            <StyledButton asIcon onClick={this.onEdit}>
              <EditIcon />
            </StyledButton>
            <StyledButton asIcon onClick={this.onDelete}>
              <DeleteIcon />
            </StyledButton>
          </span>
        ) : null}
      </Wrapper>
    );
  }
}

export default withTranslation(I18N_CORE)(RatingWrapper);
