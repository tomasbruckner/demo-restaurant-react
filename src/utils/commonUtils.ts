import format from 'date-fns/format';
import parseISO from 'date-fns/parseISO';

export const isRunningInJest = () => {
  return process.env.JEST_WORKER_ID !== undefined;
};

export const formatDate = (date: string) => {
  return date ? format(parseISO(date), 'MMM d, yyyy') : date;
};
