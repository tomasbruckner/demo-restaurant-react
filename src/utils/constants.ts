// eslint-disable-next-line import/prefer-default-export
export const AppTheme = {
  default: (Symbol('defaultAppTheme') as unknown) as string,
};

export const Role = {
  Admin: 1,
  Owner: 2,
  User: 3,
};

export const AllRoles = [Role.Admin, Role.Owner, Role.User];
export const AdminOrOwner = [Role.Admin, Role.Owner];
export const OnlyAdmin = [Role.Admin];
export const DEFAULT_PAGE_SIZE = 20;
export const DEFAULT_SEARCH_PAGE_SIZE = 10;
