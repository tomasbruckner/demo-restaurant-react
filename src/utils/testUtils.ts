export const fakeTranslate = (key: string) => key;

export const fakeLoad = () => Promise.resolve();

export const fakeLogger = {
  info() {},
  error() {},
};

// eslint-disable-next-line no-unused-vars
export const fakeFunction = (): any => {};

export const fakeHistory = {
  push: () => {},
};

export const fakeTranslationProps = {
  t: fakeTranslate,
  i18n: {} as any,
  tReady: true,
};

export const fakeReview = {
  reviewId: 1,
  reviewRatingId: 2,
  restaurantId: 3,
  title: 'title',
  text: 'text',
  dateOfVisit: '',
  hasOwnerResponse: true,
  created: '',
  owner: {
    userId: 4,
    firstName: 'firstName',
    lastName: 'lastName',
    imageLink: 'imageLink',
  },
  ownerResponse: {
    reviewReplyId: 5,
    reviewId: 6,
    creatorId: 7,
    created: '',
    updated: '',
    text: 'text',
  },
  restaurantName: 'restaurantName',
};

export const fakeUser = {
  userId: 1,
  roleId: 2,
  firstName: 'firstName',
  lastName: 'lastName',
  email: 'email@email',
  created: '',
  imageLink: 'string',
};

export const fakeRestaurant = {
  restaurantId: 1,
  ownerId: 2,
  name: 'name',
  description: 'description',
  websiteUrl: 'websiteUrl',
  contactPhone: 'contactPhone',
  address: 'address',
  numberOfReviews: 3,
  averageRating: 4,
  hasCurrentUserReviewed: true,
  imageLink: 'imageLink',
};

export const fakeRouterProps = {
  location: {} as any,
  match: {} as any,
  history: { push: () => {} } as any,
};
