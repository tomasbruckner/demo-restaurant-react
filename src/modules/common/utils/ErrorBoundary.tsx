import React, { PureComponent } from 'react';
import { WithTranslation, withTranslation } from 'react-i18next';
import Logger from 'utils/logger';
import { I18N_CORE } from 'core/i18n/i18n';
import ErrorScreen from './ErrorScreen';

export class ErrorBoundaryComp extends PureComponent<
  { renderProp?: () => React.ReactNode } & WithTranslation
> {
  state = {
    error: null,
  };

  static getDerivedStateFromError(error: Error) {
    return { error };
  }

  componentDidCatch(error: unknown, errorInfo: unknown) {
    Logger.error(error, errorInfo);
  }

  render() {
    const { children, renderProp, t } = this.props;
    const { error } = this.state;

    if (!error) {
      return children;
    }

    if (renderProp) {
      return renderProp();
    }

    return <ErrorScreen title={t('CommonComponents.ErrorBoundary.error')} />;
  }
}

export default withTranslation(I18N_CORE)(ErrorBoundaryComp);
