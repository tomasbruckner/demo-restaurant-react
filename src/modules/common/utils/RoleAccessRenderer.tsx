import React, { PureComponent, ReactNode } from 'react';
import { connect, ConnectedProps } from 'react-redux';
import { RootState } from 'core/redux/store';
import { getUserFromState } from 'core/redux/ducks/UserDuck';
import { Role } from 'utils/constants';

const connector = connect((state: RootState) => {
  return {
    user: getUserFromState(state),
  };
});

type Props = {
  onlyAdmin?: boolean;
  adminOrOwner?: boolean;
  renderNotAllowed?: ReactNode;
  children: React.ReactNode;
};

class RoleAccessRenderer extends PureComponent<Props & ConnectedProps<typeof connector>> {
  checkPermissions = () => {
    const { user, onlyAdmin, adminOrOwner } = this.props;

    if (!user) {
      return false;
    }

    if (onlyAdmin) {
      return user.roleId === Role.Admin;
    }

    if (adminOrOwner) {
      return user.roleId === Role.Admin || user.roleId === Role.Owner;
    }

    return false;
  };

  render() {
    const { children, renderNotAllowed = null } = this.props;

    if (this.checkPermissions()) {
      return children;
    }

    return renderNotAllowed;
  }
}

export default connector(RoleAccessRenderer);
