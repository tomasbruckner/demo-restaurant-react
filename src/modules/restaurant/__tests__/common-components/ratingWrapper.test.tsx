import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { RatingWrapper } from 'modules/restaurant/common-components/RatingWrapper';
import { fakeTranslationProps } from 'utils/testUtils';

describe('Rating Wrapper', () => {
  it('basic example', () => {
    const tree = renderer
      .create(
        <RatingWrapper
          text="test"
          averageRating={1}
          numberOfReviews={24}
          {...fakeTranslationProps}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
