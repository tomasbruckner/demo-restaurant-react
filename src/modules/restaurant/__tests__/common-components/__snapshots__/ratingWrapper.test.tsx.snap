// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating Wrapper basic example 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 500px;
}

<div
  className="c0"
>
  <span
    aria-label="1 Star"
    className="MuiRating-root MuiRating-sizeLarge MuiRating-readOnly"
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    role="img"
  >
    <span
      className="MuiRating-decimal"
    >
      <span
        style={
          Object {
            "overflow": "hidden",
            "position": "absolute",
            "width": "0%",
            "zIndex": 1,
          }
        }
      >
        <span
          className="MuiRating-icon MuiRating-iconFilled"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
      <span
        style={Object {}}
      >
        <span
          className="MuiRating-icon MuiRating-iconFilled"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
    </span>
    <span
      className="MuiRating-decimal"
    >
      <span
        style={
          Object {
            "overflow": "hidden",
            "position": "absolute",
            "width": "0%",
            "zIndex": 1,
          }
        }
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
      <span
        style={Object {}}
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
    </span>
    <span
      className="MuiRating-decimal"
    >
      <span
        style={
          Object {
            "overflow": "hidden",
            "position": "absolute",
            "width": "0%",
            "zIndex": 1,
          }
        }
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
      <span
        style={Object {}}
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
    </span>
    <span
      className="MuiRating-decimal"
    >
      <span
        style={
          Object {
            "overflow": "hidden",
            "position": "absolute",
            "width": "0%",
            "zIndex": 1,
          }
        }
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
      <span
        style={Object {}}
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
    </span>
    <span
      className="MuiRating-decimal"
    >
      <span
        style={
          Object {
            "overflow": "hidden",
            "position": "absolute",
            "width": "0%",
            "zIndex": 1,
          }
        }
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
      <span
        style={Object {}}
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
      </span>
    </span>
  </span>
  <h6
    className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
  >
    test
  </h6>
  <span>
    <button
      className="MuiButtonBase-root MuiIconButton-root ButtonComponent-root-1"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiIconButton-root ButtonComponent-root-1"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
    </button>
  </span>
</div>
`;
